tests/components/shelly/test_coordinator.py

```diff
@@ -0,0 +1,539 @@
+"""Tests for Shelly coordinator."""
+from datetime import timedelta
+from unittest.mock import AsyncMock
+
+from aioshelly.exceptions import DeviceConnectionError, InvalidAuthError
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.shelly.const import (
+    ATTR_CHANNEL,
+    ATTR_CLICK_TYPE,
+    ATTR_DEVICE,
+    ATTR_GENERATION,
+    DOMAIN,
+    ENTRY_RELOAD_COOLDOWN,
+    RPC_RECONNECT_INTERVAL,
+    SLEEP_PERIOD_MULTIPLIER,
+    UPDATE_PERIOD_MULTIPLIER,
+)
+from homeassistant.config_entries import SOURCE_REAUTH, ConfigEntryState
+from homeassistant.const import ATTR_DEVICE_ID, STATE_ON, STATE_UNAVAILABLE
+from homeassistant.helpers.device_registry import (
+    async_entries_for_config_entry,
+    async_get as async_get_dev_reg,
+)
+from homeassistant.util import dt
+
+from . import (
+    init_integration,
+    inject_rpc_device_event,
+    mock_polling_rpc_update,
+    mock_rest_update,
+    register_entity,
+)
+
+from tests.common import async_fire_time_changed
+
+RELAY_BLOCK_ID = 0
+LIGHT_BLOCK_ID = 2
+SENSOR_BLOCK_ID = 3
+DEVICE_BLOCK_ID = 4
+
+
+async def test_block_reload_on_cfg_change(hass, mock_block_device, monkeypatch):
+    """Test block reload on config change."""
+    await init_integration(hass, 1)
+
+    monkeypatch.setattr(mock_block_device.blocks[DEVICE_BLOCK_ID], "cfgChanged", 1)
+    mock_block_device.mock_update()
+    await hass.async_block_till_done()
+
+    # Generate config change from switch to light
+    monkeypatch.setitem(
+        mock_block_device.settings["relays"][RELAY_BLOCK_ID], "appliance_type", "light"
+    )
+    monkeypatch.setattr(mock_block_device.blocks[DEVICE_BLOCK_ID], "cfgChanged", 2)
+    mock_block_device.mock_update()
+    await hass.async_block_till_done()
+
+    assert hass.states.get("switch.test_name_channel_1") is not None
+
+    # Wait for debouncer
+    async_fire_time_changed(
+        hass, dt.utcnow() + timedelta(seconds=ENTRY_RELOAD_COOLDOWN)
+    )
+    await hass.async_block_till_done()
+
+    assert hass.states.get("switch.test_name_channel_1") is None
+
+
+async def test_block_no_reload_on_bulb_changes(hass, mock_block_device, monkeypatch):
+    """Test block no reload on bulb mode/effect change."""
+    monkeypatch.delattr(mock_block_device.blocks[RELAY_BLOCK_ID], "red")
```

Comment:
May be explain why attr-s are removed?

Reply:
We use a single mock for all tests which we adapt according to the test

Reply:
I get that, the suggestion was to add a comment to the test.

Reply:
When trying to add explanation I realized this part is not needed, I copied it from another test and forgot to remove it :flushed: